### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vsupruniuk**

### CODE REVIEW

**Comment by vsupruniuk** [comment]
image
image
image
image
image

File: taxi/views.py [hunk 1]
```diff
@@ -13,3 +14,29 @@ def index(request):
     }
 
     return render(request, "taxi/index.html", context=context)
+
+
+class ManufacturerListView(generic.ListView):
+    model = Manufacturer
+    queryset = Manufacturer.objects.all().order_by("name")
```
Comment (LiudmylaKulinchenko) [inline taxi/views.py]: It will work without `.all()`

File: taxi/views.py [hunk 2]
```diff
@@ -13,3 +14,29 @@ def index(request):
...
+    paginate_by = 5
+
+
+class CarListView(generic.ListView):
+    model = Car
+    queryset = Car.objects.all().select_related("manufacturer")
```
Comment (LiudmylaKulinchenko) [inline taxi/views.py]: same with select_related

File: taxi/views.py [hunk 3]
```diff
@@ -13,3 +14,29 @@ def index(request):
...
+
+
+class DriverListView(generic.ListView):
+    model = Driver
+    paginate_by = 5
+
+
+class DriverDetailView(generic.DetailView):
+    model = Driver
+    queryset = Driver.objects.all().prefetch_related("cars")
```
Comment (LiudmylaKulinchenko) [inline taxi/views.py]: same here

File: templates/includes/pagination.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            </span>
+        </li>
+
+        {% if page_obj.has_next %}
+            <li class="page-item">
+                <a class="page-link" href="?page={{ page_obj.next_page_number }}">Next</a>
+            </li>
+        {% endif %}
+    </ul>
+{% endif %}
```
Comment (LiudmylaKulinchenko) [inline templates/includes/pagination.html]: Add a blank line

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

**Review by LiudmylaKulinchenko — APPROVED** [review]
Great job 😌☀

**Pull request closed without merging**